### PR TITLE
Ch645 add observation notes tile

### PR DESCRIPTION
**Pull request by cquiroz**
This PR adds a placeholder for the observation notes tile.
It is mostly work on the observation layout. In particular it now supports maximizing and restoring a fixed size tile

screencast 2021-01-14 20-37-55

### CODE REVIEW

File: explore/src/main/scala/explore/tabs/ObsTabContents.scala [hunk 1]
```diff
@@ -35,46 +36,102 @@ import react.semanticui.elements.button.Button.ButtonProps
...
+      breakPointNote(BreakpointName.sm).composeLens(layoutItemHeight)
+    def notesHeightState(s: State): TileSizeState = notesHeight.getOption(s) match {
+      case Some(x) if x === 1 => TileSizeState.Minimized
+      case _                  => TileSizeState.Normal
+    }
+  }
+
+  type Props = ObsTabContents
   implicit val propsReuse: Reusability[Props] = Reusability.derive
+  implicit val stateReuse: Reusability[State] = Reusability.never
```
Comment (rpiaggio) [inline explore/src/main/scala/explore/tabs/ObsTabContents.scala:127]: Why can't we reuse the `State`?

File: common/src/main/scala/explore/components/Tile.scala [hunk 2]
```diff
@@ -3,17 +3,30 @@
...
 import react.common._
+import react.common.implicits._
 import react.semanticui.collections.menu._
+import react.semanticui.elements.button.Button
 
 final case class TileButton(body: VdomNode)
 
-final case class Tile(title: String, movable: Boolean, back: Option[TileButton])
-    extends ReactPropsWithChildren[Tile](Tile.component)
+final case class Tile(
```
Comment (rpiaggio) [inline common/src/main/scala/explore/components/Tile.scala:19]: I understand that all our tiles are unmovable now, is that right?

File: explore/src/main/scala/explore/tabs/ObsTabContents.scala (hunk 1 above)
Comment (cquiroz) [reply explore/src/main/scala/explore/tabs/ObsTabContents.scala:127]: Oh, I missed this. I had it as `never` to get it to compile but I need to make it properly derive

File: common/src/main/scala/explore/components/Tile.scala (hunk 2 above)
Comment (cquiroz) [reply common/src/main/scala/explore/components/Tile.scala:19]: I'd expect yes but we don't need to specify that in the tile but in the layout. The flag here was to put an icon to use as a handle
but that is not needed anymore

File: common/src/main/scala/explore/components/Tile.scala [hunk 3]
```diff
@@ -3,17 +3,30 @@
...
+  title:             String,
+  back:              Option[TileButton] = None,
+  canMinimize:       Boolean = false,
+  canMaximize:       Boolean = false,
+  state:             TileSizeState = TileSizeState.Normal,
+  sizeStateCallback: TileSizeState => Callback = _ => Callback.empty
```
Comment (rpiaggio) [inline common/src/main/scala/explore/components/Tile.scala:25]: These two parameters can be conmbined into a `View` (and probably use `ViewF.fromState` at the call site).

**Review by rpiaggio — APPROVED** [review]
CSS linter is complaining, otherwise LGTM

File: common/src/main/scala/explore/components/Tile.scala (hunk 3 above)
Comment (cquiroz) [reply common/src/main/scala/explore/components/Tile.scala:25]: That's a good idea, let me try that

**Pull request merged**